article/app/liveblog/LiveBlogCurrentPage.scala

```diff
@@ -1,10 +1,71 @@
 package liveblog
 
-import model.liveblog.BodyBlock
+import model.liveblog.{Blocks, BodyBlock}
+
+sealed trait BlockRange { def query: Option[Seq[String]] }
+case object Canonical extends BlockRange {
+  val firstPage = "body:latest:30"
+  val oldestPage = "body:oldest:1"
+  val timeline = "body:key-events"
+  // this only makes sense for liveblogs at the moment, but article use field body not blocks anyway
+  val query = Some(Seq(firstPage, oldestPage, timeline))
+}
+case class PageWithBlock(page: String) extends BlockRange {
+  // just get them all, the caching should prevent too many requests, could use "around"
+  val query = None
+}
+case object ArticleBlocks extends BlockRange {
+  // we currently only use this for emails, we just want all the blocks
+  val query = None
+}
+case class SinceBlockId(lastUpdate: String) extends BlockRange {
+  val around = s"body:around:$lastUpdate:5"
+  // more than 5 could come in (in one go), but unlikely and won't matter as it'll just fetch again soon
```

Comment:
@TBonnin what I mean is the endpoint is checked every 5 seconds when you are looking at the live blog, but if you leave it in the background it can increase as far as 20 minutes.  It's possible the blogger would add more than 5 posts within that time.
```
current view:

vvv-newest          oldest-vvv
"i, j, k, l, m, n, o, p, q, r"

stuff that exists in capi
a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r

next update w'nt get us up to date:
"d, e, f, g, h"

subsequent update gets us up to date:
"a, b, c"
```
ideally it would get all 8 in the update, but it's a tradeoff between requesting a lot of blocks and re-requesting.

Also, in a susequent PR I'm going to make it cache responses once they become non-empty, so it would probably be more like:
```
update 1:
h
update 2:
g
update 3:
f
....etc....
```
however at that point I'll probably make it short circuit the delay when it gets a non empty response